homeassistant/components/withings/sensor.py

```diff
@@ -462,6 +567,8 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the sensor config entry."""
+    ent_reg = er.async_get(hass)
+
```

Comment:
Unpack `hass.data[DOMAIN][entry.entry_id]` into a local var so you don't have to keep doing dict lookups here

Reply:
I have a separate PR to make it a dataclass

Reply:
https://github.com/home-assistant/core/pull/102547